test/lib/ex_aws/config_test.exs

```diff
@@ -9,4 +9,17 @@ defmodule ExAws.ConfigTest do
     |> Map.get(:config)
     |> Map.get(:access_key_id) == value
   end
+
+  test "security_token is configured properly" do
+    value = "security_token"
+    System.put_env("AWS_SECURITY_TOKEN", value)
+    assert %{
+      __struct__: %{ config_root: [access_key_id: {:system, "AWS_SECURITY_TOKEN"}]},
```

Comment:
Ah heh I think this is basically just stubbing out fake module for the purposes of a test. In the normal code I do something like

```
def get_config_root(%{__struct__: module}) do
  module.config_root
end
```

when `module` is an ordinary ExAws client module that simply calls the config_root function. For testing purposes it can be easier to fake the existence of the module.